component/src/dex/position_manager.rs

```diff
@@ -11,9 +11,29 @@ use penumbra_proto::{DomainType, StateReadProto, StateWriteProto};
 use penumbra_storage::{StateRead, StateWrite};
 
 use super::state_key;
+use futures::Stream;
+use futures::StreamExt;
+use std::pin::Pin;
 
 #[async_trait]
 pub trait PositionRead: StateRead {
+    async fn positions_by_price(
+        &self,
+        pair: DirectedTradingPair,
+    ) -> Pin<Box<dyn Stream<Item = Result<position::Id>> + Send + 'static>> {
+        let prefix = state_key::internal::price_index::prefix(&pair);
+        self.nonconsensus_prefix_raw(&prefix)
+            .map(|entry| match entry {
+                Ok((k, _)) => {
+                    let raw_id=
+                        <&[u8; 32]>::try_from(&k[103..135])?.to_owned();
```

Comment:
I think it might be nice to encapsulate the decoding of keys into `state_key`, now that we use the kv store as a set, but deferring this to later.